src/main/scala/cromwell/engine/WorkflowManager.scala

```diff
@@ -0,0 +1,89 @@
+package cromwell.engine
+
+import java.util.UUID
+
+import akka.actor.{Actor, ActorSystem, ActorRef}
+import akka.pattern.pipe
+import cromwell.util.WriteOnceStore
+import scala.concurrent.ExecutionContext.Implicits.global
+import cromwell.binding.{WdlBinding, WorkflowInputs, WdlSource}
+import cromwell.engine.WorkflowActor.Start
+import cromwell.engine.WorkflowManagerActor.{SubmitWorkflow, WorkflowStatus}
+import cromwell.engine.backend.Backend
+import cromwell.engine.backend.local.LocalBackend
+
+import scala.concurrent.Future
+import scala.util.Try
+
+/**
+ * Abstract notion of a workflow manager. Mainly exists to allow for actor-system-free unit testing of basic concepts
+ */
+trait WorkflowManager {
+  type Workflow
+
+  case class ManagedWorkflow(id: WorkflowId, workflow: Workflow)
+
+  val backend: Backend
+
+  def generateWorkflow(wdl: WdlSource, inputs: WorkflowInputs): Try[Workflow]
+  def workflowStatus(id: WorkflowId): Option[WorkflowState]
```

Comment:
Also i go back to my statement the other day that Try's main usefulness over Option is conveying multiple possible error modes. There's only one here